### ISSUE TITLE
[expo/fingerprint] Absolute paths for AAR files when using expo-location@18.0.4 causes fingerprint miscalculation

### ISSUE THREAD

**Issue by Vadko** [expo/expo#33725]
### Summary

When using `fingerprint` runtimePolicy with expo-location@18.0.4 (it works without any issues with expo-location@18.0.3) I've noticed that fingerprint generated between local and build environment is always different (when doing local build as well). After debugging fingerprints from workdir and build directory I spotted that absolute paths are used:

workdir: 

```
"aarProjects": [
    {
        "aarFilePath": "/private/var/folders/k2/130gftqn5sl74zxvzqcnch7m0000gn/T/eas-build-local-nodejs/ba85e5e5-3a1a-4d41-be44-7325fa61d6a4/build/node_modules/expo-location/android/libs/io.nlopez.smartlocation-3.3.3-jetified.aar",
        "name": "expo-location$io.nlopez.smartlocation-jetified-aar",
        "projectDir": "/private/var/folders/k2/130gftqn5sl74zxvzqcnch7m0000gn/T/eas-build-local-nodejs/ba85e5e5-3a1a-4d41-be44-7325fa61d6a4/build/node_modules/expo-location/android/build/expo-location$io.nlopez.smartlocation-jetified-aar"
    }
]
```

build dir:

```
"aarProjects": [
    {
        "aarFilePath": "/Users/vadkointel/WebstormProjects/***/node_modules/expo-location/android/libs/io.nlopez.smartlocation-3.3.3-jetified.aar",
        "name": "expo-location$io.nlopez.smartlocation-jetified-aar",
        "projectDir": "/Users/vadkointel/WebstormProjects/***/node_modules/expo-location/android/build/expo-location$io.nlopez.smartlocation-jetified-aar"
    }
]
```

### What platform(s) does this occur on?

Android

### SDK Version

52.0.19

### Environment

```text
expo-env-info 1.2.1 environment info:
    System:
      OS: macOS 15.1.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.17.0 - /usr/local/bin/node
      Yarn: 1.22.10 - ~/.volta/bin/yarn
      npm: 10.8.2 - /usr/local/bin/npm
      Watchman: 2024.10.28.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.16.2 - /Users/vadkointel/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.1, iOS 18.1, macOS 15.1, tvOS 18.1, visionOS 2.1, watchOS 11.1
    IDEs:
      Android Studio: 2024.1 AI-241.19072.14.2412.12360217
      Xcode: 16.1/16B40 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~52.0.19 => 52.0.19 
      expo-router: ~4.0.13 => 4.0.13 
      react: 18.3.1 => 18.3.1 
      react-dom: 18.3.1 => 18.3.1 
      react-native: 0.76.5 => 0.76.5 
      react-native-web: ~0.19.13 => 0.19.13 
    npmGlobalPackages:
      eas-cli: 12.6.0
    Expo Workflow: bare
```

### Minimal reproducible example

https://github.com/Vadko/location-fingerprint

`eas build -p android --local`

```
[CONFIGURE_EXPO_UPDATES] Runtime version mismatch
[CONFIGURE_EXPO_UPDATES] Runtime version on your local machine: 0cc00d4fee12453313fca521fd9875b68d96debd
[CONFIGURE_EXPO_UPDATES] Runtime version calculated on EAS: df0ef7c3c532afa4bc54252e59b7bf6665870235
```

**Comment by CristyTarantino** [comment]
Exactly the same issue here.

**Comment by Kudo** [comment]
thanks for reporting the issue. `@expo/fingerprint@0.11.6` (included in `expo@52.0.23`) should resolve the issue.
